### PR TITLE
avm2: Implement XML.parent()

### PR DESCRIPTION
**Pull request by AllinolCP**
Hi! this is my first PR to ruffle 
Please let me know what I should improve

progresses #8330

### CODE REVIEW

**Comment by Dinnerbone** [comment]
Welcome and thank you!
Can you please add a test case for when it's undefined?

**Comment by AllinolCP** [comment]
added!

**Comment by evilpie** [comment]
You can simplify your tests by adding `XML.prettyPrinting = false` at the top of your test. toString in Flash and Ruffle should then match.

**Comment by AllinolCP** [comment]
> You can simplify your tests by adding `XML.prettyPrinting = false` at the top of your test. toString in Flash and Ruffle should then match.

great idea, i will try this

File: core/src/avm2/globals/xml.rs [hunk 1]
```diff
@@ -147,6 +149,19 @@ pub fn children<'gc>(
...
 }
 
+pub fn parent<'gc>(
+    activation: &mut Activation<'_, 'gc>,
+    this: Option<Object<'gc>>,
+    _args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error<'gc>> {
+    let xml = this.unwrap().as_xml_object().unwrap();
+    let node = xml.node();
+    if node.parent().is_none() {
```
Comment (evilpie) [inline core/src/avm2/globals/xml.rs:159]: Using is_none and unwrap doesn't feel very rusty to me. You can probably use Option::map_or here.

Comment (AllinolCP) [reply core/src/avm2/globals/xml.rs]: should be resolved now!

**Review by Aaron1011 — APPROVED** [review]

**Review by Herschel — APPROVED** [review]
Thank you, and welcome!

**Pull request merged**